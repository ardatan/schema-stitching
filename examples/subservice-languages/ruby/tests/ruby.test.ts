import { ChildProcessWithoutNullStreams, spawn } from 'child_process';
import { join } from 'path';
import { gatewayApp } from '../gateway';

describe('Ruby subservices', () => {
  let servicesProcess: ChildProcessWithoutNullStreams;
  beforeAll(async () => {
    servicesProcess = spawn('npm', ['run', 'start-services'], {
      cwd: join(__dirname, '..'),
    });
  });
  afterAll(() => {
    servicesProcess.stdout.destroy();
    servicesProcess.stderr.destroy();
    servicesProcess.kill();
  });
  it('should work', async () => {
    const result = await gatewayApp.fetch('/graphql', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        query: /* GraphQL */ `
          query {
            users(ids: ["1", "2"]) {
              id
              name
              username
              reviews {
                body
                product {
                  name
                }
              }
            }
          }
        `,
      }),
    });
    const json = await result.json();
    expect(json).toMatchSnapshot('result');
  });
});
