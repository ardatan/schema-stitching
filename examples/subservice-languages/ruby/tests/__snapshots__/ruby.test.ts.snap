// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ruby subservices should work: result 1`] = `
{
  "data": {
    "users": [
      {
        "id": "1",
        "name": "Ada Lovelace",
        "reviews": [
          {
            "body": "Love it!",
            "product": {
              "name": "Table",
            },
          },
          {
            "body": "Too expensive.",
            "product": {
              "name": "Couch",
            },
          },
        ],
        "username": "@ada",
      },
      {
        "id": "2",
        "name": "Alan Turing",
        "reviews": [
          {
            "body": "Could be better.",
            "product": {
              "name": "Chair",
            },
          },
          {
            "body": "Prefer something else.",
            "product": {
              "name": "Table",
            },
          },
        ],
        "username": "@complete",
      },
    ],
  },
}
`;
